objects.go

```diff
@@ -168,6 +168,40 @@ func (o *Object) Clone() *Object {
 	return &objectCopy
 }
 
+// IsContainedInRect : returns true if the object
+// A primitive is in the rect if the entire primitive is
+// A Mesh without a primitive is in the rect if it's centre is
+func (o *Object) IsContainedInRect(rect Rect) bool {
+	var xMin, xMax, yMin, yMax float32
+
+	if o.GetPrimitive() != nil {
+		return o.GetPrimitive().IsContainedInRect(*o, rect)
```

Comment:
I'm unclear why we are doing the fine grained check here based on primitive _and_ the coarse grained check based on origin below.  I would have thought it would be important to the caller whether a fine grained or coarse grained check was performed, so we should have separate calls for each and let the caller choose?